src/Compilers/Core/Portable/CaseInsensitiveComparison.cs

```diff
@@ -14,69 +14,105 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public static class CaseInsensitiveComparison
     {
-        /// <summary>
-        /// This class seeks to perform one-to-one lowercase Unicode case mappings, which should be culture invariant.
-        /// </summary>
-        private sealed class OneToOneUnicodeComparer : StringComparer
+        // PERF: Cache the TextInfo for a neutral culture since this will be accessed very frequently
+        private static readonly TextInfo s_neutralCultureTextInfo = GetNeutralCulture().TextInfo;
+
+        private static CultureInfo GetNeutralCulture()
         {
-            // PERF: Grab the TextInfo for the invariant culture since this will be accessed very frequently
-            private static readonly TextInfo s_invariantCultureTextInfo = CultureInfo.InvariantCulture.TextInfo;
-
-            /// <summary>
-            /// ToLower implements the one-to-one Unicode lowercase mapping
-            /// as descriped in ftp://ftp.unicode.org/Public/UNIDATA/UnicodeData.txt.
-            /// The VB spec states that these mappings are used for case-insensitive
-            /// comparison
-            /// </summary>
-            /// <param name="c"></param>
-            /// <returns>If <paramref name="c"/> is upper case, then this returns its lower case equivalent. Otherwise, <paramref name="c"/> is returned unmodified.</returns>
-            public static char ToLower(char c)
+            CultureInfo culture;
+            try
             {
-                // PERF: This is a very hot code path in VB, optimize for ASCII
-
-                // Perform a range check with a single compare by using unsigned arithmetic
-                if (unchecked((uint)(c - 'A')) <= ('Z' - 'A'))
+                culture = new CultureInfo("en");
+                if (culture.IsNeutralCulture)
                 {
-                    return (char)(c | 0x20);
+                    return culture;
                 }
+            }
+            catch (ArgumentException) // System.Globalization.CultureNotFoundException not on all platforms
+            {
+                // Apparently we are running on a platform without the "en" culture.
+                // Let us try a couple of fall-backs...
+            }
 
-                if (c < 0xC0) // Covers ASCII (U+0000 - U+007F) and up to the next upper-case codepoint (Latin Capital Letter A with Grave)
-                {
-                    return c;
-                }
+            culture = CultureInfo.CurrentCulture;
+            if (culture.IsNeutralCulture)
```

Comment:
If the culture is neutral, then it has no culture-specific rules for character mapping. In that case is doesn't matter if the underlying locale it is French, Japanese, or Russian, as they will all have the same character mapping for the purposes used herein.